addons/l10n_id/models/qris_transaction.py

```diff
@@ -0,0 +1,76 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from datetime import timedelta
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class QRISTransaction(models.Model):
+    """QRIS Transaction
+
+    General table to store a certian unique transaction with QRIS details attached
+    """
+    _name = "l10n_id.qris.transaction"
+    _description = "Record of QRIS transactions"
+
+    model = fields.Char()  # payment in respond to which model
+    model_id = fields.Char()  # id/uuid
+
+    # Fields that store the QRIS details coming from API request
+    qris_invoice_id = fields.Char(readonly=True)
+    qris_amount = fields.Integer(readonly=True)
+    qris_content = fields.Char(readonly=True)
+    qris_creation_datetime = fields.Datetime(readonly=True)
+
+    bank_id = fields.Many2one("res.partner.bank", help="Bank used to generate the current QRIS transaction")
+    paid = fields.Boolean(help="Payment Status of QRIS")
+
+    @api.constrains('model')
+    def _constraint_model(self):
+        # only allow `account.move` model, to be overriden in other moduels`
+        if self.model != 'account.move':
+            raise ValidationError(_("QRIS capability is not extended to model %s yet!", self.model))
+
+    def _get_record(self):
+        """ Get the backend invoice record that the qris transaction is handling
+        To be overriden in other modules"""
+        self.ensure_one()
+        return self.env['account.move'].browse(int(self.model_id)).exists()
```

Comment:
Why would you need exists?

Reply:
Also please check for the model and ensure it is account.move, return nothing if not.
It would be safer

Reply:
I think browse() alone doesn't check if a certain record exists, with exists() if the record turns out to not exist, then it will rerturn empty recordset.

Noted on the model checking